pkg/controllers/state/cluster.go

```diff
@@ -183,6 +187,40 @@ func (c *Cluster) IsNodeNominated(providerID string) bool {
 	return false
 }
 
+// IsNodeUnderutilized returns true if the last pod event on the node was at least consolidateAfter
+// in the past. Also returns a reason to inform the user on how this was calculated.
+func (c *Cluster) IsNodeUnderutilized(providerID string, consolidateAfter time.Duration) (bool, time.Duration, string) {
```

Comment:
I wonder if rather than making the cluster state methods tightly scoped to this underutilization use-case if you just expose the last cluster event that happened on the state node. That would basically mean that you wouldn't have to pass down the consolidateAfter value into the function call and I think it should reduce the number of return arguments that you have for this function call

Reply:
Also, the check would get much quicker